.github/workflows/nodejs.yml

```diff
@@ -0,0 +1,39 @@
+name: auth-landing CI
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm ci
+      - name: Format code
+        run: npm run check-format
+      - name: Lint svelte files
+        run: npm run lint
+      - name: Generate production build
+        run: npm run export
+      - name: Add SSH client
```

Comment:
The deployment steps should be ran only on `master` branch update; these should not be ran for PRs.

Reply:
We want to run CI on both push and PR. However, we want to skip some steps conditionally. For example, deployment should only be run on `gh-pages` branch.

Reply:
Specifically, we want to build on **all** branches and build **all** PRs. Deploy to Github Pages should only be run on branch `gh-pages`. All other steps are always required.

Reply:
>  Deploy to Github Pages should only be run on branch `gh-pages`.

I think you mean 'branch `master`'.

Reply:
To summarize, the desired configuration is:
- For updates to all PRs and all branches (except `gh-pages` branch)
    - Run build steps (linting, format check, build bundle, etc.)
- For updates to `master` branch
    - Run deploy steps after build steps (push bundles to `gh-pages` branch)

Also, do you have access to repository settings? If you do, please setup a deploy key and action secrets.

Reply:
I do not have access to repo settings

Reply:
Okay, I'll setup the deploy key then.